peer.go

```diff
@@ -1177,7 +1177,19 @@ func (p *peer) handleDownStreamPkt(state *commitmentState, pkt *htlcPacket) {
 		// to our local log, then update the commitment
 		// chains.
 		htlc.ChannelPoint = state.chanPoint
-		index := state.channel.AddHTLC(htlc)
+		index, err := state.channel.AddHTLC(htlc)
+		if err != nil {
+			peerLog.Errorf("Adding HTLC rejected: %v", err)
```

Comment:
The error here should also be sent across the `htlcPacket`'s error (`pkt.err`) channel. This way the RPC caller is notified of the errors. 

However this is just a stop-gap before we decide on either if we want to do retry logic within the daemon, or force clients to perform retires themselves. 

With that said, can you add two TODO's here for me? 

 1. TODO: possibly perform fallback/retry logic depending on type of error
 2. TODO: send a cancel message back to the `htlcSwitch`. 

Thanks!